### PR TITLE
Fixed crashes.

### PR DESCRIPTION
**Pull request by teamagam3**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/teamagam/GimelGimel/pull/74" target="_blank">ReviewNinja

File: app/src/main/java/com/teamagam/gimelgimel/app/view/LauncherActivity.java [hunk 1]
```diff
@@ -58,21 +60,33 @@ public void onCreate(Bundle savedInstanceState) {
...
             LocationFetcher.askForLocationPermission(this);
+        } else {
+            requestGpsLocationUpdates();
+            startMainActivity();
         }
     }
 
+    private void requestGpsLocationUpdates() {
+        if (!mLocationFetcher.getIsRequestingUpdates()) {
+            mLocationFetcher.requestLocationUpdates(mLocationMinUpdatesMs, mLocationMinDistanceM);
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/LauncherActivity.java]: This line can throw an exception if it doesn't have any provider attached to it. (in-case all the tryAddProvider method calls failed).
try-catch that and log appropriately. !fix

File: app/src/main/java/com/teamagam/gimelgimel/app/view/LauncherActivity.java [hunk 2]
```diff
@@ -58,21 +60,33 @@ public void onCreate(Bundle savedInstanceState) {
...
+        // Request for log permissions before we use it
+        if (!doesHaveGpsPermissions()) {
             LocationFetcher.askForLocationPermission(this);
+        } else {
+            requestGpsLocationUpdates();
+            startMainActivity();
         }
     }
 
+    private void requestGpsLocationUpdates() {
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/LauncherActivity.java]: re-organize file method order. (private methods after public methods) !fix

Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/LauncherActivity.java]: !fixed

File: app/src/main/java/com/teamagam/gimelgimel/app/view/LauncherActivity.java (hunk 1 above)
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/LauncherActivity.java]: !fixed

**Pull request merged**